File: MSMA_Backend/src/main/java/com/spring/constants/UserType.java
package com.spring.constants;

import com.spring.exceptions.BusinessException;
import lombok.Getter;

@Getter
public enum UserType {
    ADMIN(0),
    USER(1),
    ARTIST(2),;

    private final int code;
    UserType(int code) {
        this.code = code;
    }
    public static String getName(int code) {
        for (UserType userType : UserType.values()) {
            if (userType.code == code) {
                return userType.name();
            }
        }
        throw new BusinessException(ApiResponseCode.INVALID_TYPE);
    }
}

File: MSMA_App/app/src/main/java/vn/edu/usth/msma/utils/constants/UserType.kt
package vn.edu.usth.msma.utils.constants

enum class UserType(val code: Int) {
    ADMIN(0),
    USER(1),
    ARTIST(2);

    companion object {
        fun getName(code: Int): String {
            return UserType.entries.find { it.code == code }?.name
                ?: throw IllegalArgumentException("Invalid code: $code")
        }
    }
}
